community/blog/2022-04 Interesting new hooks and .NET 6 support in Arcus WebAPI v1.5/README.md

```diff
@@ -0,0 +1,143 @@
+
+While this is a very helpful feature, the customization was missing something crucial: we don't allow custom sanitization of the request/response body. This, too, can contain sensitive or personal information that should not be exposed in any logging system. Arcus Web API v1.5 solved this the same way the sanitization of headers occurs: with overidable functionality.
+
+Here a rudimentary(!) example of what this could look like. Note that all overidable functionality is optional and doesn't have to be overridden.
+
+```csharp
+using Arcus.WebApi.Logging;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+public class PIIRequestTrackingMiddleware : RequestTrackingMiddleware
+{
+    public PIIRequestTrackingMiddleware(RequestDelegate next, ILogger<RequestTrackingMiddleware> logger) : base(next, logger)
+    {
+    }
+
+    protected override string SanitizeRequestBody(HttpRequest request, string requestBody)
+    {
+        if (request.HasJsonContentType())
+        {
+            var json = JObject.Parse(requestBody);
+            json.Property("clientId").Remove();
+
+            return json.ToString();
+        }
+
+        return requestBody;
+    }
+
+    protected override string SanitizeResponseBody(HttpResponse response, string responseBody)
+    {
+        if (request.HasJsonContentType())
+        {
+            var json = JObject.Parse(responseBody);
+            json.Property("clientName").Remove();
+
+            return json.ToString();
+        }
+
+        return responseBody;
+    }
+}
+```
+
+This custom-made request tracking middleware can then be given to the registration of the request tracking so that next time a request is processed, the personal and/or sensitive information is filtered-out from the tracking.
+
+```csharp
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+
+public class Startup
+{
+    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+    {
+        app.UseRouting();
+
+        app.UseRequestTracking<PIIRequestTrackingMiddleware>();
+
+        ...
+        app.UseMvc();
+    }
+}
+```
+
+## Built-in JSON formatting extensions
+Our [Arcus Web API project template](https://templates.arcus-azure.net/features/web-api-template) is using several great and highly recommended approaches to start off your project. One of these approaches is to limit the input/output formatting of models to only JSON formatting. What this means is that the registered input/output formatters were inspected and altered. While this is a valuable addition to the Web API project template, it is still somewhat boilerplate code that with its inspection code doesn't make the template a very user-friendly or clean place to work in.
+
+Arcus Web API v1.5 changed this. In a new library called `Arcus.WebApi.Hosting`, we have added extensions on the MVC options so that this JSON formatting limitation can be added in a more fluent manner.
```

Comment:
```suggestion
Arcus Web API v1.5 changes this. In a new library called `Arcus.WebApi.Hosting`, we have added extensions on the MVC options so that this JSON formatting limitation can be added in a more fluent manner.
```